src/Orleans.Core.Abstractions/Async/PublicOrleansTaskExtensions.cs

```diff
@@ -0,0 +1,168 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Orleans
+{
+    /// <summary>
+    /// Utility functions for dealing with Tasks.
+    /// </summary>
+    public static class PublicOrleansTaskExtensions
```

Comment:
Let's not move these to Abstractions. They are not Orleans related, they are just helpers, and we have been guilty of changing these and breaking people that use binding redirects in the past already.
Also we have some potential plans for codegen to avoid these much boxing that we do.

If some method is required by generated code, we can discuss ways to avoid it. It might be fine to let it go for beta, but not for the final release.